### PR TITLE
couple of memory leaks fixes.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by devnexen** [comment]
sorry for the noise. new PR

**Comment by est31** [comment]
No problem.

File: src/util/srp.cpp [hunk 1]
```diff
@@ -542,7 +542,7 @@ static SRP_Result fill_buff()
 
 	if (!fp) return SRP_ERR;
 
-	if (fread(g_rand_buff, sizeof(g_rand_buff), 1, fp) != 1) return SRP_ERR;
+	if (fread(g_rand_buff, sizeof(g_rand_buff), 1, fp) != 1) { fclose(fp); return SRP_ERR; }
```
Comment (est31) [inline src/util/srp.cpp]: Reminder to myself: backport this to csrp-gmp.

Comment (nerzhul) [inline src/util/srp.cpp]: it's why std::unique_ptr is very useful on c++ types :)

**Comment by nerzhul** [comment]
Okay for me :+1:

**Comment by sfan5** [comment]
:+1:

**Comment by nerzhul** [comment]
Merged as 48b3bb980d4a026d32739acc1982f16e3c303c5b thanks

**Pull request closed without merging**